homeassistant/components/nederlandse_spoorwegen/sensor.py

```diff
@@ -47,13 +48,17 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the departure sensor."""
 
     nsapi = ns_api.NSAPI(config[CONF_API_KEY])
+
     try:
         stations = nsapi.get_stations()
     except (
         requests.exceptions.ConnectionError,
         requests.exceptions.HTTPError,
     ) as error:
-        _LOGGER.error("Couldn't fetch stations, API key correct?: %s", error)
+        _LOGGER.error("Could not connect to the internet: %s", error)
+        return
```

Comment:
We can `raise PlatformNotReady` here to let the platform setup retry later.

Reply:
Like this? Or both exceptions?

Reply:
We only want to retry on recoverable errors, eg internet down. Bad credentials is not recoverable, but requires the user to change the config and set up again.